driver/src/main/java/org/neo4j/driver/internal/security/TLSSocketChannel.java

```diff
@@ -345,18 +345,22 @@ private HandshakeStatus wrap( ByteBuffer buffer ) throws IOException, ClientExce
             }
             else
             {
+                logger.debug( "Network output buffer doesn't need enlarging, flushing data to the channel instead to open up space on the buffer." );
                 // flush as much data as possible
                 cipherOut.flip();
-                if ( channelWrite( cipherOut ) == 0 )
-                {
-                    throw new ClientException( format(
-                                    "Failed to enlarge network buffer from %s to %s. This is either because the " +
-                                    "new size is however less than the old size, or because the application " +
-                                    "buffer size %s is so big that the application data still cannot fit into the " +
-                                    "new network buffer.", curNetSize, netSize, buffer.capacity() ) );
+                while ( cipherOut.hasRemaining() ) {
```

Comment:
Yeah, I also thought on adding a limit but;

If we end up in a position where we could not write anything on a socket, it most probably points a resource starvation on the computer either cpu or network buffers related. In either case the problem lies somewhere else and if we throw some exception here - where theoretically it is not an error - we would break the runtime and eventually become a point of failure.

We also loop on the same condition at [line](https://github.com/neo4j/neo4j-java-driver/pull/389/files/a78e441dc69486fc5b7649f1d11e396b5ace3d9f#diff-4bff1912d58fd3cac510c55ab49334e1R329) on the same method, so I though this would also not be a problem.

Does it make sense?